src/main/java/com/hazelcast/aws/impl/DescribeInstances.java

```diff
@@ -56,14 +58,9 @@
     private Map<String, String> attributes = new HashMap<String, String>();
 
     public DescribeInstances(AwsConfig awsConfig, String endpoint) throws IOException {
-        if (awsConfig == null) {
-            throw new IllegalArgumentException("AwsConfig is required!");
-        }
         this.awsConfig = awsConfig;
         this.endpoint = endpoint;
 
-        checkKeysFromIamRoles(new Environment());

```

Comment:
It was there to check ECS credentials from `Environment` and only to be used when it's running in ECS but it was being initiated with every call to `checkKeysFromIamRoles` even it's not necessary. I've moved the initiation to `fillKeysFromIamTaskRole` method to prevent unnecessary object creation.